api/clouddrive.proto

```diff
@@ -0,0 +1,232 @@
+//
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "google/api/annotations.proto";
+import "api/api.proto";
+
+package openstorage.api;
+
+option go_package = "api";
+option java_multiple_files = true;
+option java_package = "com.openstorage.api";
+
+// CloudDriveSet defines a set of drive pools that could be attached on a node
+message CloudDriveSet {
+	// ID is the unique id of the cloud drive. This is managed by the storage provider
+  // implementing the handlers
+	string id = 1;
+	// Pools describes the configuration of the drives present in this set
+	repeated StoragePool pools = 2;
+	// CreateTimestamp is the timestamp when the drive set was created
+  google.protobuf.Timestamp create_timestamp = 3;
+	// State state of the drive set from the well defined states
+	CloudDriveSetState state = 4;
+	// SchedulerNodeName is the name of the node in scheduler context who is using
+  // the cloud drive set
+	string scheduler_node_name = 5;
+	// InstanceID is the cloud provider id of the instance using this drive set
+	string instance_id = 6;
+	// Zone defines the zone in which the node exists
+	string zone = 7;
+}
+
+
+// CloudDriveSetState indicates the state of the DriveSet
+enum CloudDriveSetState {
+	// Initializing indicates that the DriveSet is attached to a node but still in
+  // init state
+	initializing = 0;
+	// InUse indicates that the DriveSet is in use
+	in_use = 1;
+	// PendingDecommission indicates that the DriveSet needs to be deleted.
+	// TODO: Currently this is being used only for storage less drive sets
+	pending_decommission = 2;
+}
+
+// OpenStorageCloudDrives is a service used to manage cloud drives in the cluster
+service OpenStorageCloudDrives {
+  // Enumerate returns the ids of all the cloud drives in the current cluster
+  rpc Enumerate(SdkCloudDriveSetEnumerateRequest)
+    returns (SdkCloudDriveSetEnumerateResponse) {
+      option(google.api.http) = {
+        get: "/v1/clouddrives"
+      };
+    }
+
+  // Create creates a new drive set
+  rpc Create(SdkCloudDriveSetCreateRequest)
+    returns (SdkCloudDriveSetCreateResponse) {
+      option(google.api.http) = {
+        post: "/v1/clouddrives"
+        body: "*"
+      };
+    }
+
+  // Expand expands the given drive set
+  rpc Expand(SdkCloudDriveSetExpandRequest)
+    returns (SdkCloudDriveSetExpandResponse) {
+      option(google.api.http) = {
+        put: "/v1/clouddrives/{id}"
+        body: "*"
+      };
+    }
+
+  // Delete deletes the given drive set
+  rpc Delete(SdkCloudDriveSetDeleteRequest)
+    returns (SdkCloudDriveSetDeleteResponse) {
+      option(google.api.http) = {
+        delete: "/v1/clouddrives/{id}"
+      };
+    }
+
+  // Delete inspects the given drive set
+  rpc Inspect(SdkCloudDriveSetInspectRequest)
+    returns (SdkCloudDriveSetInspectResponse) {
+      option(google.api.http) = {
+        get: "/v1/clouddrives/{id}"
+      };
+    }
+
+  // Attach attaches all the drives in the given drive set
```

Comment:
nit: Attach attaches all the drives in the given drive set to a single instance provided in the request